services/worker/src/worker/job_runners/config/parquet_metadata.py

```diff
@@ -0,0 +1,185 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+from functools import lru_cache, partial
+from typing import List, Optional, TypedDict
+
+from huggingface_hub import HfFileSystem
+from huggingface_hub.hf_file_system import safe_quote
+from libcommon.constants import (
+    PARQUET_REVISION,
+    PROCESSING_STEP_CONFIG_PARQUET_METADATA_VERSION,
+)
+from libcommon.exceptions import (
+    FileSystemError,
+    ParquetResponseEmptyError,
+    PreviousStepFormatError,
+)
+from libcommon.processing_graph import ProcessingStep
+from libcommon.storage import StrPath
+from libcommon.utils import JobInfo
+from libcommon.viewer_utils.parquet_metadata import create_parquet_metadata_file
+from pyarrow.parquet import ParquetFile
+from tqdm.contrib.concurrent import thread_map
+
+from worker.config import AppConfig
+from worker.job_runners.config.config_job_runner import ConfigJobRunner
+from worker.job_runners.config.parquet_and_info import ParquetFileItem
+from worker.utils import CompleteJobResult, get_previous_step_or_raise
+
+
+class ParquetFileAndMetadataItem(TypedDict):
+    dataset: str
+    config: str
+    split: str
+    url: str
+    filename: str
+    size: int
+    num_rows: int
+    parquet_metadata_subpath: str
+
+
+class ConfigParquetMetadataResponse(TypedDict):
+    parquet_files_and_metadata: List[ParquetFileAndMetadataItem]
+
+
+@lru_cache(maxsize=128)
+def get_hf_fs(hf_token: Optional[str]) -> HfFileSystem:
+    """Get the Hugging Face filesystem.
+
+    Args:
+        hf_token (Optional[str]): The token to access the filesystem.
+    Returns:
+        HfFileSystem: The Hugging Face filesystem.
+    """
+    return HfFileSystem(token=hf_token)
```

Comment:
we have to pass `endpoint`, if I'm not mistaken. Otherwise we cannot use the Hub CI for e2e tests